### PR TITLE
Application grids

### PR DESCRIPTION
**Pull request by IvarStefansson**
## Proposed changes

We start using create_mdg in ModelGeometry and provide some predefined geometries in the application folder for use with or without models.

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [ ] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [ ] The documentation is up-to-date.
- [ ] Static typing is included in the update.
- [ ] This PR does not duplicate existing functionality.
- [ ] The update is covered by the test suite (including tests added in the PR).

### CODE REVIEW

**Comment by IvarStefansson** [comment]
Could you please have a second look, @jhabriel, before passing on to code owners? In particular, I'd like your feedback on the functions in mdg_library. Other changes since last review include cleaning up in applications and a tiny first step towards roll-out in tests of models, which I don't intend to pursue any further in this PR.

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:1]:
```suggestion
"""Library of mixed-dimensional grids.

This module is primarily envisioned as a library of mixed-dimensional grids
```

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:19]:
```suggestion
    meshing_args: dict[str, float],
```

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:50]:
```suggestion
    fracture_network = pp.create_fracture_network(fractures, domain)
```

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:88]:
```suggestion
    fracture_network = pp.create_fracture_network(fractures, domain)
```

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:52]: I would not return the domain here, since one can always access it via `fracture_network.domain`.

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:90]: Same comment as above.

File: tests/functional/setups/manu_flow_incomp_frac.py
Comment (jhabriel) [inline tests/functional/setups/manu_flow_incomp_frac.py:1]: After merging #863, I will open a PR with a heavy refactoring of the functional test setups. I recommend reverting the changes of this file and `tests/functional/setups/manu_poromech_nofrac.py`.

**Review by jhabriel — CHANGES_REQUESTED** [review]
Great, only minor comments and a few suggested changes.

**Comment by IvarStefansson** [comment]
> Great, only minor comments and a few suggested changes.

Thanks. I think I may have been a bit unclear when saying
>  I'd like your feedback on the functions in mdg_library.
Could you please weigh in on whether we need the functions at all? I'm unsure about whether they might not create more entropy than they are worth. @keileg, this question goes to you as well.

File: src/porepy/applications/md_grids/mdg_library.py
Comment (IvarStefansson) [reply src/porepy/applications/md_grids/mdg_library.py:52]: Fair enough.

File: tests/functional/setups/manu_flow_incomp_frac.py
Comment (IvarStefansson) [reply tests/functional/setups/manu_flow_incomp_frac.py:1]: Won't work, since the tests won't run and I'll not be able to merge.

File: src/porepy/applications/md_grids/domains.py
Comment (OmarDuran) [inline src/porepy/applications/md_grids/domains.py:22]: Perhaps box?

File: src/porepy/fracs/fracture_network.py
Comment (OmarDuran) [inline src/porepy/fracs/fracture_network.py:14]: Is this modification associated with mypy?

File: tests/integration/models/setup_utils.py
Comment (OmarDuran) [inline tests/integration/models/setup_utils.py:51]: As a matter of information, by setting `"cell_size_min": 0.5 * ls`, the cell size will be `0.5 * ls`.

**Review by OmarDuran — APPROVED** [review]
It looks good, I left some minor comments. thanks for the standardization task.

File: src/porepy/fracs/fracture_network.py
Comment (IvarStefansson) [reply src/porepy/fracs/fracture_network.py:14]: Yes

File: examples/mandel_biot.py
Comment (jhabriel) [inline examples/mandel_biot.py:1255]:
```suggestion
        """Set meshing arguments."""
```

File: examples/mandel_biot.py
Comment (jhabriel) [inline examples/mandel_biot.py:1257]:
```suggestion
        default_meshing_arguments = {"cell_size": 2 * ls}
```

File: examples/mandel_biot.py
Comment (jhabriel) [inline examples/mandel_biot.py:1258]:
```suggestion
        return self.params.get("meshing_arguments", default_meshing_arguments)
```

File: examples/mandel_biot.py
Comment (jhabriel) [inline examples/mandel_biot.py:1260]:
```suggestion
    def grid_type(self) -> Literal["simplex", "cartesian", "tensor_grid"]:
```

File: examples/mandel_biot.py
Comment (jhabriel) [inline examples/mandel_biot.py:1254]:
```suggestion
    def meshing_arguments(self) -> dict[str, float]:
```

File: examples/terzaghi_biot.py
Comment (jhabriel) [inline examples/terzaghi_biot.py:521]:
```suggestion
    def meshing_arguments(self) -> dict[str, float]:
```

File: src/porepy/__init__.py
Comment (jhabriel) [inline src/porepy/__init__.py:113]: Why do we need to include the `Fracture` class in the top-level init? This is an abstract class that _should_ never be instantiated. Right?

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:2]:
```suggestion

Mainly for use in tests. Other usage should be covered by the model_geometries.
```

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:57]:
```suggestion
) -> tuple[pp.MixedDimensionalGrid, pp.FractureNetwork3d]:
```

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:21]:
```suggestion
) -> tuple[pp.MixedDimensionalGrid, pp.FractureNetwork2d]:
```

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:8]:
```suggestion
from typing import Literal
```

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:66]:
```suggestion
        **kwargs: Keyword arguments for meshing as used by :meth:`~porepy.grids.mdg_generation.create_mdg`.
```

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:30]:
```suggestion
        **kwargs: Keyword arguments for meshing as used by :meth:`~porepy.grids.mdg_generation.create_mdg`.
```

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:26]:
```suggestion
        meshing_args: Keyword arguments for meshing as used by :meth:`~porepy.grids.mdg_generation.create_mdg`.
```

File: src/porepy/applications/md_grids/mdg_library.py
Comment (jhabriel) [inline src/porepy/applications/md_grids/mdg_library.py:22]: Description should fit in only one line. A more detailed description can go below.

File: src/porepy/fracs/fracture_network.py
Comment (jhabriel) [inline src/porepy/fracs/fracture_network.py:7]:
```suggestion
from typing import Optional, Union
```

File: src/porepy/fracs/fracture_network.py
Comment (jhabriel) [inline src/porepy/fracs/fracture_network.py:14]:
```suggestion
FractureList = Union[list[pp.LineFracture], List[pp.PlaneFracture]]
```

**Comment by IvarStefansson** [comment]
> It looks good, I left some minor comments. thanks for the standardization task.

Thanks for the swift review!

File: src/porepy/models/geometry.py
Comment (jhabriel) [inline src/porepy/models/geometry.py:132]:
```suggestion
            Meshing arguments compatible with :meth:`~porepy.grids.mdg_generation.create_mdg`.
```

**Review by jhabriel — APPROVED** [review]
Super! I left a few docstring/typing related suggestions.

One thing though. Can you check whether is necessary to include the `Fracture` class in the top-level init. My understanding is that this is an abstract class, and should be therefore not be part of the top level init. Could you check this before merging please?

File: src/porepy/__init__.py
Comment (IvarStefansson) [reply src/porepy/__init__.py:113]: I had a reason for it, but I think you're right.

File: src/porepy/applications/md_grids/mdg_library.py
Comment (IvarStefansson) [reply src/porepy/applications/md_grids/mdg_library.py:22]: I don't think that's a strict rule. At least, we haven't enforced it strictly in the past.

File: src/porepy/fracs/fracture_network.py
Comment (IvarStefansson) [reply src/porepy/fracs/fracture_network.py:14]: I tried that first. For some reason, Python 3.8 refused, so I had to revert.

**Pull request merged**
